src/components/Hexagons/HexagonGrid.svelte

```diff
@@ -0,0 +1,44 @@
+<script lang="ts">
+  // Inspired by https://github.com/sveltejs/svelte-virtual-list/blob/master/VirtualList.svelte
+  type T = $$Generic;
+
+  export let items: T[];
+  export let cols: number;
+
+  export let gap: 'small' | 'medium' | 'big' = 'medium';
+</script>
+
+<!-- Warning: Very magic, totally eyeballed numbers in the next line -->
+<ul
+  class="grid gap-{gap}"
+  style="grid-template-columns: 0.2425fr repeat({cols}, 0.4805fr 0.2425fr); grid-auto-rows: 0.5fr 0.5fr;"
+>
+  {#each items as item, index}
+    {@const col = index % cols}
+    {@const row = (index - col) / cols}
+    {@const colStart = 2 * col + 1}
+    {@const colEnd = 2 * col + 4}
+    {@const rowStart = 2 * row + 1 + (col % 2)}
+    {@const rowEnd = 2 * row + 3 + (col % 2)}
+    <li style="grid-column: {colStart} / {colEnd}; grid-row: {rowStart} / {rowEnd}">
+      <slot {item} {col} {row} />
```

Comment:
I'm trying to thing how the slot can use the col and row, do you have an example?

Reply:
I was thinking maybe conditional styling based on the column or row. Or even skipping the rendering of certain items based on the column and/or row. Do you think I should remove that?

Reply:
I think that makes sense but it's not super clear, maybe a comment would help